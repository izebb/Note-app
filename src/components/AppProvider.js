import React, { createContext, Component } from "react";
import { withRouter } from "react-router-dom";

import { Notes } from "../utils/notes";

const AppContext = React.createContext();

export class Provider extends Component {
  constructor(props) {
    super(props);

    this.notes = new Notes();

    this.state = {
      notes: this.getNotes(),
      textEditorInput: "",
      noteTitle: "",
      searchValue: "",
      getNoteById: this.getNoteById,
      onSearch: this.onSearch,
      createNote: this.createNote,
      onTitleChange: this.onTitleChange,
      onEditorChange: this.onEditorChange,
      deleteNote: this.deleteNote,
      updateNote: this.updateNote,
      addNote: this.addNote
    };
  }

  getNotes() {
    return [...this.notes.get()].reverse();
  }

  deleteNote = id => {
    this.notes.delete(id);
    const notes = this.getNotes();

    this.setState({ notes }, this.redirectToNote);
  };

  updateNote = id => {
    const { textEditorInput, noteTitle } = this.state;

    if (noteTitle.trim()) {
      this.notes.update(id, {
        noteTitle,
        textEditorInput
      });

      this.setState({
        notes: this.getNotes()
      });
    }
  };

  createNote = () => {
    const { textEditorInput, noteTitle } = this.state;
    const id = this.notes.create({
      noteTitle: "New note",
      textEditorInput: ""
    });

    this.setState(
      {
        searchValue: "",
        notes: this.getNotes()
      },
      () => this.redirectToNote(id)
    );
  };

  redirectToNote = (id = "") => {
    const { history } = this.props;

    history.push(`/${id}`);
  };

  onEditorChange = e => {
    this.setState({
      textEditorInput: e.target.value
    });
  };

  onSearch = e => {
    const value = e.target.value;

    this.setState({
      searchValue: e.target.value,
      notes: this.notes.search(value.trim())
    });
  };

  onTitleChange = e => {
    this.setState({
      noteTitle: e.target.value
    });
  };

  getNoteById = id => {
    const item = this.notes.get(id);

    if (item) {
      this.setState(item);
    } else {
      this.redirectToNote();
    }
  };

  render() {
    return (
      <AppContext.Provider value={{ ...this.state }} {...this.props}>
        {this.props.children}
      </AppContext.Provider>
    );
  }
}

const AppConsumer = AppContext.Consumer;
const AppProvider = withRouter(Provider);

export { AppContext, AppConsumer, AppProvider };
