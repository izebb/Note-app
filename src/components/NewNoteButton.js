import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";

import { NotesList } from "./NotesList";

const NewNoteStyled = styled.div`
  width: 100%;
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const NewButtonStyled = styled.button`
  color: #fff;
  background-color: #6021f3;
  padding: 0 50px;
  height: 37px;
  font-size: 13px;
  border-radius: 5px;
  height: 37px;
  font-size: 13px;
  border-radius: 50px;
  box-shadow: 1px 1px 10px rgba(0, 0, 0, 0.32);
  border: none;
  font-weight: 600;
  text-decoration: none;
  display: flex;
  align-items: center;
  &:focus {
    text-decoration: none;
  }
`;

export function NewNoteButton({ onClick }) {
  return (
    <NewNoteStyled>
      <NewButtonStyled onClick={onClick}>Add Note</NewButtonStyled>
    </NewNoteStyled>
  );
}

NewNoteButton.defaultProps = {
  onClick: () => {}
};
