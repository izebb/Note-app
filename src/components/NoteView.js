import React, { Fragment, useContext, useEffect } from "react";
import styled from "styled-components";

import { AppContext } from "./AppProvider";
import { TextEditor } from "./TextEditor";

const ViewPaperContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  border: 1px solid #e6e6e6;
  position: relative;
  background-color: rgba(244, 255, 254, 0.78);

  &:before {
    content: " ";
    position: absolute;
    width: 100%;
    height: 100%;
    border: 1px solid #efefef;
    top: 0;
    z-index: -1;
    box-shadow: 1px 2px 10px #f6fffe;
  }
`;

const ViewTitleBarStyled = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const NoteTitleStyled = styled.input`
  width: 100%;
  height: 58px;
  padding: 7px 10px;
  font-size: 20px;
  font-weight: 700;
  background: transparent;
  border: none;
  border-bottom: 1px solid #90b7b4;
  &:focus {
    outline: none;
    border-bottom: 1px solid #5f7d8c;
  }
`;

export function NoteView({ match }) {
  const {
    saveNote,
    noteTitle,
    onTitleChange,
    textEditorInput,
    onEditorChange,
    getNoteById,
    deleteNote,
    updateNote
  } = useContext(AppContext);
  const { id = "" } = match.params;

  useEffect(() => {
    if (id) {
      getNoteById(id);
    }
  }, [id]);

  return (
    <ViewPaperContainer>
      <ViewTitleBarStyled>
        <NoteTitleStyled
          onBlur={() => updateNote(id)}
          value={noteTitle}
          placeholder="Title of note"
          onChange={onTitleChange}
        />
      </ViewTitleBarStyled>
      <TextEditor
        save={() => updateNote(id)}
        onEditorChange={onEditorChange}
        textEditorInput={textEditorInput}
      />
    </ViewPaperContainer>
  );
}
