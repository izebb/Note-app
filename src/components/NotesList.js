import React, { useContext } from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";
import dateFormat from "dateFormat";

import binIcon from "../images/bin.svg";
import { AppContext } from "./AppProvider";

const NoteListContainerStyled = styled.div`
  flex: 1;
  overflow-y: auto;
`;

const DeleteButton = styled.button`
  background-color: transparent;
  border: none;
  width: 40px;
  position: absolute;
  right: 5px;
  opacity: 0;
  transform: translateX(20px);
  transition: transform 200ms ease-out, opacity 150ms ease-out;
`;

const NoteItemStyled = styled.div`
  display: flex;
  flex-direction: column;
  padding: 15px 20px 12px 20px;
  margin-left: 40px;
  color: #ccc;
  border-bottom: 1px dashed #b5b5b5;
  text-decoration: none;
  position: relative;
  padding-right: 45px;
  overflow: hidden;

  &:hover {
    ${DeleteButton} {
      opacity: 0.8;
      transform: none;
    }
  }
  &.selected {
    color: #fff;
  }
`;

const NoteItemTitleStyled = styled.div`
  font-weight: 700;
`;

const NoteItemDateStyled = styled.div`
  font-size: 12px;
  margin-top: 20px;
`;

const SearchContainerStyled = styled.div`
  padding: 0 30px;
  margin: 10px 0;
`;

const SearchStyled = styled.input`
  width: 100%;
  height: 30px;
  box-sizing: border-box;
  border: none;
  padding: 0 5px;
  border-radius: 4px;
  font-size: 14px;

  &:focus {
    outline: none;
  }
`;

export function NotesList({ searchValue, onSearch, data, onClick, onDelete }) {
  const removeNote = id => () => {
    if (confirm("Do you really wanna delete this note")) {
      onDelete(id);
    }
  };

  return (
    <NoteListContainerStyled>
      <SearchContainerStyled>
        <SearchStyled
          value={searchValue}
          onChange={onSearch}
          placeholder="Search Notes..."
        />
      </SearchContainerStyled>
      {data.map(item => {
        return (
          <NoteItemStyled
            className="note"
            as={NavLink}
            key={item.id}
            to={`/${item.id}`}
            activeClassName="selected"
          >
            <NoteItemTitleStyled>{item.noteTitle}</NoteItemTitleStyled>
            <NoteItemDateStyled>
              {dateFormat(item.createdAt, "mmmm dS, yyyy, h:MM:ss TT")}
            </NoteItemDateStyled>
            <DeleteButton id="delete_button" onClick={removeNote(item.id)}>
              <img src={binIcon} />
            </DeleteButton>
          </NoteItemStyled>
        );
      })}
    </NoteListContainerStyled>
  );
}

NotesList.defaultProps = {
  data: [],
  onClick: () => {},
  onDelete: () => {},
  onSearch: () => {}
};
