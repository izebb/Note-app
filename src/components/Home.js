import React, { useContext } from "react";
import { Redirect } from "react-router-dom";
import styled from "styled-components";
import { AppContext } from "./AppProvider";
import { NewNoteButton } from "./NewNoteButton";

import journalImage from "../images/journal.svg";

const ImgStyled = styled.img`
  height: 240px;
  opacity: 0.35;
`;

const TextStyled = styled.p`
  color: #444;
  margin-bottom: 20px;
`;

const EmptyContainerStyled = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 1;
  font-weight: 300;
  font-size: 30px;
  padding-top: 130px;
  color: #444;
`;

export function Home() {
  const { notes, createNote } = useContext(AppContext);
  const item = notes[0];

  if (item) {
    return <Redirect to={`/${item.id}`} replace />;
  }

  return (
    <EmptyContainerStyled>
      <ImgStyled src={journalImage} />
      <TextStyled>You do not have any notes. Let's make one for you</TextStyled>
      <div>
        <NewNoteButton onClick={createNote} />
      </div>
    </EmptyContainerStyled>
  );
}
