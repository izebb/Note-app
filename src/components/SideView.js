import React, { useContext } from "react";
import styled from "styled-components";

import { AppContext } from "./AppProvider";
import { NotesList } from "./NotesList";
import { NewNoteButton } from "./NewNoteButton";

const SideViewStyled = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 300px;
  height: 100%;
  background-color: #576cff;
  padding: 15px 0 0 0;
  display: flex;
  flex-direction: column;
`;

export function SideView() {
  const { notes, createNote, deleteNote, searchValue, onSearch } = useContext(
    AppContext
  );

  return (
    <SideViewStyled>
      <NewNoteButton onClick={createNote} />
      <NotesList
        data={notes}
        onDelete={deleteNote}
        onSearch={onSearch}
        searchValue={searchValue}
      />
    </SideViewStyled>
  );
}
