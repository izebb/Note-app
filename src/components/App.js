import React from "react";
import styled from "styled-components";
import { BrowserRouter as Router, Route } from "react-router-dom";

import { SideView } from "./SideView";
import { NoteView } from "./NoteView";
import { AppProvider } from "./AppProvider";
import { Home } from "./Home";

const AppContainerStyled = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  min-width: 1200px;
  width: 100%;
  height: 100%;
`;

const MainViewStyled = styled.div`
  height: 100%;
  margin-left: 300px;
  display: flex;
`;

const MainViewContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  max-width: 750px;
  margin: 30px auto;
`;

export function App() {
  return (
    <AppContainerStyled>
      <Router>
        <AppProvider>
          <SideView />
          <MainViewStyled>
            <MainViewContainer>
              <Route path="/" exact component={Home} />
              <Route path="/:id" component={NoteView} />
            </MainViewContainer>
          </MainViewStyled>
        </AppProvider>
      </Router>
    </AppContainerStyled>
  );
}
