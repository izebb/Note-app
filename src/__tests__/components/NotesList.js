import React from "react";
import { MemoryRouter as Router } from "react-router-dom";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { NotesList } from "../../components/NotesList";

window.confirm = jest.fn().mockReturnValue(true);

describe("NotesList", () => {
  const props = {
    onSearch: jest.fn(),
    onClick: jest.fn(),
    onDelete: jest.fn(),
    data: [
      {
        id: "71c34d00-6357-11e9-8524-59c0881df9c7",
        createdAt: 1555756283344,
        noteTitle: "New note",
        textEditorInput: "sample text"
      }
    ],
    searchValue: ""
  };

  it("should render without errors", () => {
    const wrapper = mount(
      <Router keyLength={0}>
        <NotesList {...props} />
      </Router>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call onDelete", () => {
    const wrapper = mount(
      <Router keyLength={0}>
        <NotesList {...props} />
      </Router>
    );

    wrapper.find("button#delete_button").simulate("click");
    expect(props.onDelete).toHaveBeenCalledWith(
      "71c34d00-6357-11e9-8524-59c0881df9c7"
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
