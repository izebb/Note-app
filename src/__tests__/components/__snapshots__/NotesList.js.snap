// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList should call onDelete 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotesList
      data={
        Array [
          Object {
            "createdAt": 1555756283344,
            "id": "71c34d00-6357-11e9-8524-59c0881df9c7",
            "noteTitle": "New note",
            "textEditorInput": "sample text",
          },
        ]
      }
      onClick={[MockFunction]}
      onDelete={
        [MockFunction] {
          "calls": Array [
            Array [
              "71c34d00-6357-11e9-8524-59c0881df9c7",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      onSearch={[MockFunction]}
      searchValue=""
    >
      <NotesList__NoteListContainerStyled>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "NotesList__NoteListContainerStyled-sc-8hxduq-0",
                "isStatic": true,
                "lastClassName": "SPOQt",
                "rules": Array [
                  "flex:1;overflow-y:auto;",
                ],
              },
              "displayName": "NotesList__NoteListContainerStyled",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "NotesList__NoteListContainerStyled-sc-8hxduq-0",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="NotesList__NoteListContainerStyled-sc-8hxduq-0 SPOQt"
          >
            <NotesList__SearchContainerStyled>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "NotesList__SearchContainerStyled-sc-8hxduq-5",
                      "isStatic": true,
                      "lastClassName": "bLwNSJ",
                      "rules": Array [
                        "padding:0 30px;margin:10px 0;",
                      ],
                    },
                    "displayName": "NotesList__SearchContainerStyled",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "NotesList__SearchContainerStyled-sc-8hxduq-5",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="NotesList__SearchContainerStyled-sc-8hxduq-5 bLwNSJ"
                >
                  <NotesList__SearchStyled
                    onChange={[MockFunction]}
                    placeholder="Search Notes..."
                    value=""
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "NotesList__SearchStyled-sc-8hxduq-6",
                            "isStatic": true,
                            "lastClassName": "bUrlEr",
                            "rules": Array [
                              "width:100%;height:30px;box-sizing:border-box;border:none;padding:0 5px;border-radius:4px;font-size:14px;&:focus{outline:none;}",
                            ],
                          },
                          "displayName": "NotesList__SearchStyled",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "NotesList__SearchStyled-sc-8hxduq-6",
                          "target": "input",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      onChange={[MockFunction]}
                      placeholder="Search Notes..."
                      value=""
                    >
                      <input
                        className="NotesList__SearchStyled-sc-8hxduq-6 bUrlEr"
                        onChange={[MockFunction]}
                        placeholder="Search Notes..."
                        value=""
                      />
                    </StyledComponent>
                  </NotesList__SearchStyled>
                </div>
              </StyledComponent>
            </NotesList__SearchContainerStyled>
            <NotesList__NoteItemStyled
              activeClassName="selected"
              as={[Function]}
              className="note"
              key="71c34d00-6357-11e9-8524-59c0881df9c7"
              to="/71c34d00-6357-11e9-8524-59c0881df9c7"
            >
              <StyledComponent
                activeClassName="selected"
                as={[Function]}
                className="note"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "NotesList__NoteItemStyled-sc-8hxduq-2",
                      "isStatic": true,
                      "lastClassName": "bpBOyh",
                      "rules": Array [
                        "display:flex;flex-direction:column;padding:15px 20px 12px 20px;margin-left:40px;color:#ccc;border-bottom:1px dashed #b5b5b5;text-decoration:none;position:relative;padding-right:45px;overflow:hidden;&:hover{",
                        ".NotesList__DeleteButton-sc-8hxduq-1",
                        "{opacity:0.8;transform:none;}}&.selected{color:#fff;}",
                      ],
                    },
                    "displayName": "NotesList__NoteItemStyled",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "NotesList__NoteItemStyled-sc-8hxduq-2",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                to="/71c34d00-6357-11e9-8524-59c0881df9c7"
              >
                <NavLink
                  activeClassName="selected"
                  className="note NotesList__NoteItemStyled-sc-8hxduq-2 bpBOyh"
                  to="/71c34d00-6357-11e9-8524-59c0881df9c7"
                >
                  <Route
                    path="\\\\/71c34d00-6357-11e9-8524-59c0881df9c7"
                  >
                    <Link
                      aria-current={null}
                      className="note NotesList__NoteItemStyled-sc-8hxduq-2 bpBOyh"
                      to="/71c34d00-6357-11e9-8524-59c0881df9c7"
                    >
                      <a
                        aria-current={null}
                        className="note NotesList__NoteItemStyled-sc-8hxduq-2 bpBOyh"
                        href="/71c34d00-6357-11e9-8524-59c0881df9c7"
                        onClick={[Function]}
                      >
                        <NotesList__NoteItemTitleStyled>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "NotesList__NoteItemTitleStyled-sc-8hxduq-3",
                                  "isStatic": true,
                                  "lastClassName": "hhiMFx",
                                  "rules": Array [
                                    "font-weight:700;",
                                  ],
                                },
                                "displayName": "NotesList__NoteItemTitleStyled",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "NotesList__NoteItemTitleStyled-sc-8hxduq-3",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="NotesList__NoteItemTitleStyled-sc-8hxduq-3 hhiMFx"
                            >
                              New note
                            </div>
                          </StyledComponent>
                        </NotesList__NoteItemTitleStyled>
                        <NotesList__NoteItemDateStyled>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "NotesList__NoteItemDateStyled-sc-8hxduq-4",
                                  "isStatic": true,
                                  "lastClassName": "dhlkpA",
                                  "rules": Array [
                                    "font-size:12px;margin-top:20px;",
                                  ],
                                },
                                "displayName": "NotesList__NoteItemDateStyled",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "NotesList__NoteItemDateStyled-sc-8hxduq-4",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="NotesList__NoteItemDateStyled-sc-8hxduq-4 dhlkpA"
                            >
                              April 20th, 2019, 12:31:23 PM
                            </div>
                          </StyledComponent>
                        </NotesList__NoteItemDateStyled>
                        <NotesList__DeleteButton
                          id="delete_button"
                          onClick={[Function]}
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "NotesList__DeleteButton-sc-8hxduq-1",
                                  "isStatic": true,
                                  "lastClassName": "eXrCde",
                                  "rules": Array [
                                    "background-color:transparent;border:none;width:40px;position:absolute;right:5px;opacity:0;transform:translateX(20px);transition:transform 200ms ease-out,opacity 150ms ease-out;",
                                  ],
                                },
                                "displayName": "NotesList__DeleteButton",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "NotesList__DeleteButton-sc-8hxduq-1",
                                "target": "button",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            id="delete_button"
                            onClick={[Function]}
                          >
                            <button
                              className="NotesList__DeleteButton-sc-8hxduq-1 eXrCde"
                              id="delete_button"
                              onClick={[Function]}
                            >
                              <img
                                src={
                                  Object {
                                    "process": [Function],
                                  }
                                }
                              />
                            </button>
                          </StyledComponent>
                        </NotesList__DeleteButton>
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </StyledComponent>
            </NotesList__NoteItemStyled>
          </div>
        </StyledComponent>
      </NotesList__NoteListContainerStyled>
    </NotesList>
  </Router>
</MemoryRouter>
`;

exports[`NotesList should render without errors 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotesList
      data={
        Array [
          Object {
            "createdAt": 1555756283344,
            "id": "71c34d00-6357-11e9-8524-59c0881df9c7",
            "noteTitle": "New note",
            "textEditorInput": "sample text",
          },
        ]
      }
      onClick={[MockFunction]}
      onDelete={[MockFunction]}
      onSearch={[MockFunction]}
      searchValue=""
    >
      <NotesList__NoteListContainerStyled>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "NotesList__NoteListContainerStyled-sc-8hxduq-0",
                "isStatic": true,
                "lastClassName": "SPOQt",
                "rules": Array [
                  "flex:1;overflow-y:auto;",
                ],
              },
              "displayName": "NotesList__NoteListContainerStyled",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "NotesList__NoteListContainerStyled-sc-8hxduq-0",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="NotesList__NoteListContainerStyled-sc-8hxduq-0 SPOQt"
          >
            <NotesList__SearchContainerStyled>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "NotesList__SearchContainerStyled-sc-8hxduq-5",
                      "isStatic": true,
                      "lastClassName": "bLwNSJ",
                      "rules": Array [
                        "padding:0 30px;margin:10px 0;",
                      ],
                    },
                    "displayName": "NotesList__SearchContainerStyled",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "NotesList__SearchContainerStyled-sc-8hxduq-5",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="NotesList__SearchContainerStyled-sc-8hxduq-5 bLwNSJ"
                >
                  <NotesList__SearchStyled
                    onChange={[MockFunction]}
                    placeholder="Search Notes..."
                    value=""
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "NotesList__SearchStyled-sc-8hxduq-6",
                            "isStatic": true,
                            "lastClassName": "bUrlEr",
                            "rules": Array [
                              "width:100%;height:30px;box-sizing:border-box;border:none;padding:0 5px;border-radius:4px;font-size:14px;&:focus{outline:none;}",
                            ],
                          },
                          "displayName": "NotesList__SearchStyled",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "NotesList__SearchStyled-sc-8hxduq-6",
                          "target": "input",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      onChange={[MockFunction]}
                      placeholder="Search Notes..."
                      value=""
                    >
                      <input
                        className="NotesList__SearchStyled-sc-8hxduq-6 bUrlEr"
                        onChange={[MockFunction]}
                        placeholder="Search Notes..."
                        value=""
                      />
                    </StyledComponent>
                  </NotesList__SearchStyled>
                </div>
              </StyledComponent>
            </NotesList__SearchContainerStyled>
            <NotesList__NoteItemStyled
              activeClassName="selected"
              as={[Function]}
              className="note"
              key="71c34d00-6357-11e9-8524-59c0881df9c7"
              to="/71c34d00-6357-11e9-8524-59c0881df9c7"
            >
              <StyledComponent
                activeClassName="selected"
                as={[Function]}
                className="note"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "NotesList__NoteItemStyled-sc-8hxduq-2",
                      "isStatic": true,
                      "lastClassName": "bpBOyh",
                      "rules": Array [
                        "display:flex;flex-direction:column;padding:15px 20px 12px 20px;margin-left:40px;color:#ccc;border-bottom:1px dashed #b5b5b5;text-decoration:none;position:relative;padding-right:45px;overflow:hidden;&:hover{",
                        ".NotesList__DeleteButton-sc-8hxduq-1",
                        "{opacity:0.8;transform:none;}}&.selected{color:#fff;}",
                      ],
                    },
                    "displayName": "NotesList__NoteItemStyled",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "NotesList__NoteItemStyled-sc-8hxduq-2",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                to="/71c34d00-6357-11e9-8524-59c0881df9c7"
              >
                <NavLink
                  activeClassName="selected"
                  className="note NotesList__NoteItemStyled-sc-8hxduq-2 bpBOyh"
                  to="/71c34d00-6357-11e9-8524-59c0881df9c7"
                >
                  <Route
                    path="\\\\/71c34d00-6357-11e9-8524-59c0881df9c7"
                  >
                    <Link
                      aria-current={null}
                      className="note NotesList__NoteItemStyled-sc-8hxduq-2 bpBOyh"
                      to="/71c34d00-6357-11e9-8524-59c0881df9c7"
                    >
                      <a
                        aria-current={null}
                        className="note NotesList__NoteItemStyled-sc-8hxduq-2 bpBOyh"
                        href="/71c34d00-6357-11e9-8524-59c0881df9c7"
                        onClick={[Function]}
                      >
                        <NotesList__NoteItemTitleStyled>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "NotesList__NoteItemTitleStyled-sc-8hxduq-3",
                                  "isStatic": true,
                                  "lastClassName": "hhiMFx",
                                  "rules": Array [
                                    "font-weight:700;",
                                  ],
                                },
                                "displayName": "NotesList__NoteItemTitleStyled",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "NotesList__NoteItemTitleStyled-sc-8hxduq-3",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="NotesList__NoteItemTitleStyled-sc-8hxduq-3 hhiMFx"
                            >
                              New note
                            </div>
                          </StyledComponent>
                        </NotesList__NoteItemTitleStyled>
                        <NotesList__NoteItemDateStyled>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "NotesList__NoteItemDateStyled-sc-8hxduq-4",
                                  "isStatic": true,
                                  "lastClassName": "dhlkpA",
                                  "rules": Array [
                                    "font-size:12px;margin-top:20px;",
                                  ],
                                },
                                "displayName": "NotesList__NoteItemDateStyled",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "NotesList__NoteItemDateStyled-sc-8hxduq-4",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <div
                              className="NotesList__NoteItemDateStyled-sc-8hxduq-4 dhlkpA"
                            >
                              April 20th, 2019, 12:31:23 PM
                            </div>
                          </StyledComponent>
                        </NotesList__NoteItemDateStyled>
                        <NotesList__DeleteButton
                          id="delete_button"
                          onClick={[Function]}
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "NotesList__DeleteButton-sc-8hxduq-1",
                                  "isStatic": true,
                                  "lastClassName": "eXrCde",
                                  "rules": Array [
                                    "background-color:transparent;border:none;width:40px;position:absolute;right:5px;opacity:0;transform:translateX(20px);transition:transform 200ms ease-out,opacity 150ms ease-out;",
                                  ],
                                },
                                "displayName": "NotesList__DeleteButton",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "NotesList__DeleteButton-sc-8hxduq-1",
                                "target": "button",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            id="delete_button"
                            onClick={[Function]}
                          >
                            <button
                              className="NotesList__DeleteButton-sc-8hxduq-1 eXrCde"
                              id="delete_button"
                              onClick={[Function]}
                            >
                              <img
                                src={
                                  Object {
                                    "process": [Function],
                                  }
                                }
                              />
                            </button>
                          </StyledComponent>
                        </NotesList__DeleteButton>
                      </a>
                    </Link>
                  </Route>
                </NavLink>
              </StyledComponent>
            </NotesList__NoteItemStyled>
          </div>
        </StyledComponent>
      </NotesList__NoteListContainerStyled>
    </NotesList>
  </Router>
</MemoryRouter>
`;
